### PR TITLE
Parse Comment & Abuse responses of the Discussion API

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

Parse the API response for comment being posted.

## Why?

Eeduce the complexity of the possible responses

Part of #8745

## Screenshots

Identical

### CODE REVIEW

File: dotcom-rendering/src/types/discussion.ts [hunk 1]
```diff
@@ -141,11 +145,60 @@ export interface CommentType {
...
+	message: string(),
+});
+
+const commentResponseSchema = variant('status', [
+	commentErrorSchema,
+	object({
+		status: literal('ok'),
+		// response.errorCode is the id of the comment that was created on the server
+		// it is returned as a string, so we need to cast to an number to be compatible
+		message: transform(string(), (input) => parseInt(input, 10)),
```
Comment (JamieB-gu) [inline dotcom-rendering/src/types/discussion.ts:180]: What if `NaN`?

File: dotcom-rendering/src/types/discussion.ts [hunk 2]
```diff
@@ -141,11 +145,60 @@ export interface CommentType {
...
+	'COMMENT_RATE_LIMIT_EXCEEDED',
+	'INVALID_PROTOCOL',
+	'AUTH_COOKIE_INVALID',
+	'READ-ONLY-MODE',
+	'API_CORS_BLOCKED',
+	'API_ERROR',
+	'EMAIL_VERIFIED',
+	'EMAIL_VERIFIED_FAIL',
+	'EMAIL_NOT_VALIDATED',
+] as const;
```
Comment (JamieB-gu) [inline dotcom-rendering/src/types/discussion.ts:166]: 👏

**Review by JamieB-gu — APPROVED** [review]

File: dotcom-rendering/src/types/discussion.ts (hunk 2 above)
Comment (mxdvl) [reply dotcom-rendering/src/types/discussion.ts:166]: There’s actually a lot more, but we’re not responding to them in any meaningful way

File: dotcom-rendering/src/types/discussion.ts (hunk 1 above)
Comment (mxdvl) [reply dotcom-rendering/src/types/discussion.ts]: Fixed in 9641ca55acc31b9f9346fc2a0408200af22d7f43

**Pull request merged**